src/main/java/duke/Task.java

```diff
@@ -0,0 +1,47 @@
+package duke;
+
+/**
+ * Task is a general class that consists of the task description and a boolean to indicate if the task is done.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructor for Task.
+     *
+     * @param description Description of the Task to do.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Constructor for Task.
+     *
+     * @param description Description of the Task to do.
+     * @param isDone      Boolean to indicate if the Task is done.
+     */
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public void markAsUndone() {
+        this.isDone = false;
+    }
```

Comment:
I think this can be combined into one function
```suggestion
    public void toggleDone() {
        this.isDone = !this.isDone;
    }
```